### PR TITLE
Fixes Space Dragon Attacking

### LINKED ISSUE #78953: Dragons won't eat or break walls
Reporting client version: 514.1589

## Round ID:
[216558](https://scrubby.melonmesa.com/round/216558)

## Testmerges:
- [Invisibility refactor](https://www.github.com/tgstation/tgstation/pull/78908)

## Reproduction:

Got the midround dragon roll in round 216558. I couldn't break down walls, clicking them just did nothing. Additionally, I was entirely unable to eat corpses. Clicking on them just attacked them.
Rifts, firebreathing, gust, and regular attacks all worked fine.

### PR DESCRIPTION
**Pull request by san7890**
Fixes #78953

## About The Pull Request

Basically the gist is that Space Dragon's special attack code was on `AttackingTarget()` rather than whatever the hell simple animals controlled by clients use (I didn't bother enough to look into the chain to remember this). This was the complete wrong proc to use, and it NEVER got executed. Anyways, we just hook into the signal for whatever the simple animal proc is as well as clean up all the code, make everything pretty, and most importantly:

MAKE THE DAMN CODE WORK
## Why It's Good For The Game

Either someone did not test their code at all, or some weird esoteric change in the attack chain changed this somehow? I'm not sure when or why this happened but it is guaranteed to be fixed now.

The code cleanup and tinkering I did means that it's gonna be about 10% easier to port this over to a basic mob eventually (not doing a full refactor when this shit is this broken, the code added here is modular enough to the point where it's plug-n-play).
## Changelog
:cl:
fix: Space Dragons can now, once again, tear down walls and eat corpses. They also have regained their special damage modifier when attacking mechs.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/hostile/space_dragon.dm [hunk 1]
```diff
@@ -121,41 +122,65 @@
...
 	if(isliving(target)) //Swallows corpses like a snake to regain health.
-		var/mob/living/L = target
-		if(L.stat == DEAD)
-			to_chat(src, span_warning("You begin to swallow [L] whole..."))
-			if(do_after(src, 30, target = L))
-				if(eat(L))
-					adjustHealth(-L.maxHealth * 0.25)
-			return
-	. = ..()
-	if(ismecha(target))
```
Comment (san7890) [inline code/modules/mob/living/simple_animal/hostile/space_dragon.dm:156]: i am so happy this got murdered

**Review by Jacquerel — APPROVED** [review]

**Comment by IndieanaJones** [comment]
I mean, that code worked for several years, and its what all megafauna used at some point. Seems like to me someone changed stuff around and didn't update everything to work properly.

**Comment by Jacquerel** [comment]
> I mean, that code worked for several years. Seems like to me someone changed stuff around and didn't update everything to work properly.

that would presumably be why the pr body says "either it didnt work or someone broke it"

**Comment by SyncIt21** [comment]
#78966 seems to be approaching this differently

**Pull request merged**